design/authn-oidc.md

```diff
@@ -0,0 +1,250 @@
+# Solution Design - Authn-OIDC/MFA Support in CLI
+
+## Table of Contents
+
+- [Solution Design - Authn-OIDC Support in CLI](#solution-design---authn-oidcmfa-support-in-cli)
+  - [Table of Contents](#table-of-contents)
+  - [Useful Links](#useful-links)
+  - [Overview](#overview)
+    - [Motivation](#motivation)
+  - [Solution](#solution)
+    - [User Experience](#user-experience)
+      - [Feature Details](#feature-details)
+      - [Scope](#scope)
+  - [Performance](#performance)
+  - [Backwards Compatibility](#backwards-compatibility)
+  - [Affected Components](#affected-components)
+  - [Test Plan](#test-plan)
+    - [Test Cases](#test-cases)
+      - [Unit Tests](#unit-tests)
+      - [E2E tests](#e2e-tests)
+      - [Security testing](#security-testing)
+  - [Logs](#logs)
+  - [Documentation](#documentation)
+  - [Version update](#version-update)
+  - [Security](#security)
+  - [Audit](#audit)
+  - [Solution Review](#solution-review)
+
+## Useful Links
+
+| Link | Private |
+|------|:-------:|
+| [Conjur Docs: OIDC Authentication](https://docs.conjur.org/Latest/en/Content/OIDC/OIDC.htm) | No |
+| [Solution Design: Authn-LDAP Support in CLI](authn-ldap.md) | No |
+| [MFA for Conjur UI - Architecture](https://ca-il-confluence.il.cyber-ark.com/display/rndp/MFA+for+Conjur+UI+-+Architecture) | Yes |
+| [MFA Customer Documentation Changes](https://ca-il-confluence.il.cyber-ark.com/display/rndp/MFA+Customer+Documentation+Changes+-+Dev+Content) | Yes |
+
+## Overview
+
+Conjur supports various forms of authentication, including OIDC (OpenID Connect). OIDC is unique in that requires the user to
+navigate to a third party website (SSO provider) and authenticate there, often using MFA. The user then receives a token from
+the SSO provider and passes this to Conjur, which verifies that the token is signed correctly by the trust SSO provider.
+This document describes the design and implementation of the OIDC authentication support in the new Python-based Conjur CLI.
+
+### Motivation
+
+OIDC support was recently added to the Conjur UI (Enterprise only), initially to support SSO with Okta. We want to extend this
+support to the CLI so developers using the CLI can authenticate to Conjur in the CLI with OIDC as they will in the Conjur UI.
+
+## Solution
+
+### User Experience
+
+The `conjur init` command should support OIDC authentication as an authentication type. It should be possible to specify
+the authentication type `oidc` using the existing `-t` or `--authn-type` option.
+When using `oidc`, a `--service-id` option should be mandatory.
+
+*Open question*: Currently, if the `--service-id` option is specified, then `--authn-type` defaults to `ldap`. Should we make this required
+without a default value instead?
```

Comment:
@jSimonCA Can we get your opinion on this as PO?

Reply:
Reviewed the solution design and workflow/customer experience looks good. Few questions/feedback from my end:
1. How is token refresh scenario handled? This is something that is getting redefined with 12.8 for MFA UI implementation so maybe we should align user experience for CLI also?
2. Please ensure that state, nonce, and code_challenge (PKCE - Proof Key for Code Exchange) are included as per security findings that came up last week. I see Jason has already mentioned it as part of his feedback as well.